src/matchPath.js

```diff
@@ -45,6 +45,10 @@ function matchPathBase(end, routePath, urlPath) {
   }
 
   const params = Object.create(null);
+  if (parentParams) {
+    Object.assign(params, parentParams);
```

Comment:
But if do not drop params from previous route then at the end of routing list params object can be too heavy and contain params from all previous routes.

Maybe merge params of parent route with params of children routes only if the parent route has flag `mergeParams: true`?